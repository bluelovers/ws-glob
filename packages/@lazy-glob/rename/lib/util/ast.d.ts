import { IPartAstInterface, IQuestionMark, IAsterisk } from '../types';
export declare function isAsterisk(g: any | IPartAstInterface): g is IAsterisk;
export declare function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark;
