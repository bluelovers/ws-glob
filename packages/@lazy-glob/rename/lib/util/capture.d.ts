export declare function capture(names: string[], glob: string): import("./globGroupsCollection").globGroupsCollectionInterface[];
