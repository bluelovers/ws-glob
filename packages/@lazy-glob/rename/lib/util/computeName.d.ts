/**
 * Builds new names based on provided glob patterns.
 * @param filenames {String[] | string} List of original names, or a single name (string), to compute new names for
 * @param srcGlob {String} Glob pattern used to match the original names
 * @param dstGlob {String} Glob pattern used to contruct new names
 * @return {String[]} List of new names. String is empty if new name could not be computed.
 * @throws {Error} An Error object
 */
export declare function computeName(filenames: string | string[], srcGlob: string, dstGlob: string): string[];
