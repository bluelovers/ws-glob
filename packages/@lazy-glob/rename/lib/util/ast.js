"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isQuestionMark = exports.isAsterisk = void 0;
const types_1 = require("../types");
function isAsterisk(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.Asterisk);
}
exports.isAsterisk = isAsterisk;
function isQuestionMark(g) {
    return (g.type === 'wildcard' && g.pattern === types_1.EnumWildcard.QuestionMark);
}
exports.isQuestionMark = isQuestionMark;
//# sourceMappingURL=ast.js.map