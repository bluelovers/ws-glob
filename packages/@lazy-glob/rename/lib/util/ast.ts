import { IPartAstInterface, IQuestionMark, EnumWildcard, IAsterisk } from '../types';

export function isAsterisk(g: any | IPartAstInterface): g is IAsterisk
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.Asterisk);
}

export function isQuestionMark(g: any | IPartAstInterface): g is IQuestionMark
{
	return (g.type === 'wildcard' && g.pattern === EnumWildcard.QuestionMark);
}
