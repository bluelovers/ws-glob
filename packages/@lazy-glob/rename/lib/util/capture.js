"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * For each string in names array, return a globGroupsCollection (an object implementing globGroupsCollectionInterface) containing the substring matched by each part of the glob pattern.
 * Example of parts of a glob: "?omer.*" => Parts: ["?"] ["omer."] ["*"]
 * @method capture
 * @param names {String[]} List of names
 * @param glob {String} The glob pattern to be applied to names
 * @return {globGroupsCollection[] | null} An array of globGroupsCollection objects. Returns null if invalid parameters or names list is empty.
 */
const globGroupsCollection_1 = require("./globGroupsCollection");
function capture(names, glob) {
    if (!Array.isArray(names) || !names.length) {
        return null;
    }
    if ((typeof glob !== 'string') || !glob.length) {
        return null;
    }
    // For each name received, create an object of type 'globGroupsCollection' containing the match for each part of the glob
    return names.map(function buildGlobGroupsCollection(name) {
        let groupsObj = globGroupsCollection_1.globGroupsCollectionFactory();
        groupsObj.initGroups(glob);
        groupsObj.buildGroups(name);
        return groupsObj;
    });
}
//# sourceMappingURL=capture.js.map