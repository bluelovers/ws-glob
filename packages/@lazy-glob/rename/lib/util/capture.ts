/**
 * For each string in names array, return a globGroupsCollection (an object implementing globGroupsCollectionInterface) containing the substring matched by each part of the glob pattern.
 * Example of parts of a glob: "?omer.*" => Parts: ["?"] ["omer."] ["*"]
 * @method capture
 * @param names {String[]} List of names
 * @param glob {String} The glob pattern to be applied to names
 * @return {globGroupsCollection[] | null} An array of globGroupsCollection objects. Returns null if invalid parameters or names list is empty.
 */
import { globGroupsCollectionFactory } from './globGroupsCollection';

function capture(names: string[], glob: string)
{
	if (!Array.isArray(names) || !names.length)
	{
		return null;
	}

	if ((typeof glob !== 'string') || !glob.length)
	{
		return null;
	}

	// For each name received, create an object of type 'globGroupsCollection' containing the match for each part of the glob
	return names.map(function buildGlobGroupsCollection(name)
	{
		let groupsObj = globGroupsCollectionFactory();
		groupsObj.initGroups(glob);
		groupsObj.buildGroups(name);
		return groupsObj;
	});
}
