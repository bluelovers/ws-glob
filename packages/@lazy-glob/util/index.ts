/**
 * Created by user on 2020/6/9.
 */

export * from './lib/types';

export default exports as typeof import('./index');
