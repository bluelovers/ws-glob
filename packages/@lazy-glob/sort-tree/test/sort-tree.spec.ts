import { inspect } from 'util'
import sortTree from '../index';

describe(`no empty`, () =>
{
	[
		[
			'packages/@yarn-tool/ws-changed/lib/',
			'packages/@yarn-tool/ws-changed/',
			'packages/ws-pkg-list/lib/',
			'packages/@yarn-tool/find-root/',
		],
	].forEach(ls =>
	{

		test(inspect(ls), () =>
		{
			const len = ls.length;

			let expected = ls.slice();
			let actual = sortTree(ls);

			expect(actual).toHaveLength(len);
			expect(actual).toContain(expected);
			expect(actual).toMatchSnapshot();

		});

	})

})
