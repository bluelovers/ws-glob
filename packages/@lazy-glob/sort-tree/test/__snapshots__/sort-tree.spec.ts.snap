// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no empty [
  'packages/@yarn-tool/ws-changed/lib/',
  'packages/@yarn-tool/ws-changed/',
  'packages/ws-pkg-list/lib/',
  'packages/@yarn-tool/find-root/'
] 1`] = `
[
  "packages/@yarn-tool/find-root/",
  "packages/@yarn-tool/ws-changed/",
  "packages/@yarn-tool/ws-changed/lib/",
  "packages/ws-pkg-list/lib/",
]
`;
